mne/beamformer/_compute_beamformer.py

```diff
@@ -154,6 +154,16 @@ def _check_proj_match(info, filters):
                              'calculating the spatial filter.')
 
 
+def _check_src_type(filters):
+    """Check whether src_type is in filters and set custom warning."""
+    if 'src_type' not in filters.keys():
```

Comment:
never ever use:

`'foo' in d.keys()`

when d is a dict. It makes it super slow as you now search in a a list-like object...

it should be:

`if 'src_type' not in filters:`